src/js/models/EventEmitter.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * The event emitter is a simple class that allows pub/sub
+ * on another class
+ */
+export default class EventEmitter {
+  constructor() {
+    this.__events = {}
+  }
+
+  /**
+   * Emit the event to all listeners
+   *
+   * @param {string}    name   The name
+   * @param {object={}} [args] Arguments to pass to
+   */
+  emit(name, args = {}) {
+    // safeguard
+    if (!name) {
+      return
+    }
+
+    // run through the event handlers and call
+    // it for each one
+    const handlers = this.__events[name]
+    if (handlers && handlers.length) {
+      for (let i = 0, handler; (handler = handlers[i]); i++) {
+        // kick each handler onto the event loop so that no one blocks the emission
+        // of the all of the events
+        // consider: deep clone args so that one handler cannot modify.
+        // This should be fine if args are simple properties
+        ;(function(h) {
```

Comment:
Sorry, I don't follow the syntax here.  Perhaps you can be a little more verbose.  This is a lambda function?